apps/emqx/src/emqx_packet.erl

```diff
@@ -66,6 +66,14 @@
 -type subscribe() :: #mqtt_packet_subscribe{}.
 -type unsubscribe() :: #mqtt_packet_unsubscribe{}.
 -type payload_encode() :: hex | text | hidden.
+-type payload_format_opts() :: #{
+    %% required option
+    payload_encode := payload_encode(),
+    %% default is ?MAX_PAYLOAD_FORMAT_SIZE
+    payload_limit => non_neg_integer(),
+    %% default is ?TRUNCATED_PAYLOAD_SIZE
+    printed_size => non_neg_integer()
```

Comment:
```suggestion
    %% Truncate if over this limit, default is ?MAX_PAYLOAD_FORMAT_SIZE
    truncate_if => non_neg_integer(),
    %% Truncate to this number of bytes, default is ?TRUNCATED_PAYLOAD_SIZE
    truncate_to => non_neg_integer()
```

Reply:
Nit: maybe `truncate_if` could be `truncate_above` (since it's an upper limit)?

Reply:
I just renamed it in the emqx_packet module. For the emqx_tracer and emqx_mgmt_api_trace modules, I kept the naming as payload_limit. It seems like a more reasonable name for the tracer HTTP API.